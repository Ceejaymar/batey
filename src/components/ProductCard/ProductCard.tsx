import { Link } from 'react-router-dom';
import styled from 'styled-components';

type ProductProps = {
  product: {
    productId: number;
    name: string;
    description: string;
    images: string[];
    price: number;
  };
  textColor?: 'white';
};

interface CardProps {
  textColor?: 'white';
}

const Card = styled(Link).withConfig({
  shouldForwardProp: (prop) => prop != 'textColor',
})<CardProps>`
  flex-basis: 1;
  text-decoration: none;
  color: ${({ textColor }) => (textColor === 'white' ? 'white' : '#284941')};

  & img {
    margin-bottom: 0.5rem;
  }

  & div {
    display: flex;
    justify-content: space-between;
    font-size: 0.8rem;
    font-weight: 500;
  }
`;

function ProductCard({ product, textColor }: ProductProps) {
  return (
    <Card
      to={`/product/${product.productId}`}
      state={{ product }}
      textColor={textColor}
    >
      <img src={product.images[0]} alt="Image of product" />

      <div>
        <p>{product.name}</p>
        <p>${product.price}</p>
      </div>
    </Card>
  );
}
export default ProductCard;
