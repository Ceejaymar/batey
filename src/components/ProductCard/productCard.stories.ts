import type { Meta, StoryObj } from '@storybook/react';
import ProductCard from './ProductCard';

const exampleProduct = {
  productId: 1,
  name: 'Product 1',
  description: 'Description of product 1',
  images: ['https://via.placeholder.com/150'],
  price: 100,
};

const meta = {
  title: 'ProductCard',
  component: ProductCard,
} satisfies Meta<typeof ProductCard>;

export default meta;

type Story = StoryObj<typeof ProductCard>;

export const Default: Story = {
  args: {
    product: exampleProduct,
  },
};
