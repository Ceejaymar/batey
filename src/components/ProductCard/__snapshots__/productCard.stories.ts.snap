// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Default smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "name": "Image of product Product 1 $100",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
  ],
  "name": "productcard--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`components/ProductCard Default smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "name": "Image of product Product 1 $100",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
  ],
  "name": "components-productcard--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
